WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.kt

```diff
@@ -144,6 +163,27 @@ class PostsListActivity : AppCompatActivity(),
                 }
             }
         })
+
+        if (site.isWPCom) {
+            // Non REST sites can't filter by author
+            viewModel.filterAuthorListItems.observe(this, Observer { items ->
+                items?.let {
+                    authorSelectionAdapter.updateItems(items)
+                    val selectionIndex = items.indexOfFirst { item -> item.isSelected }
+                    if (selectionIndex >= 0) {
+                        authorSelection.setSelection(selectionIndex)
+                    }
+                }
+            })
+        }
+
+        viewModel.filterOnlyUser.observe(this, Observer { onlyUser ->
+            onlyUser?.let {
+                postsPagerAdapter = PostsPagerAdapter(POST_LIST_PAGES, site, onlyUser, supportFragmentManager)
```

Comment:
Wouldn't we lose the user's position if we re-create the whole adapter like this without handling the currently selected tab?

Reply:
I am also not sure about our desired behavior for when we select a different author filter, do we want to completely refresh the page, or just the items. We can leave this question for now though and handle it in a separate PR if necessary.

Reply:
I've updated the code to retain the position through adapter changes for now while we decide how we want to update those various fragments.